src/main/ruby/core/truffle/cext.rb

```diff
@@ -127,6 +127,15 @@ def data_holder
   class RbEncoding
     NAME_FIELD_INDEX = 0
 
+    @cache = {}
+    @mutex = Mutex.new
+
+    def self.get(encoding)
+      @mutex.synchronize do
+        @cache.fetch(encoding) { |key| @cache[key] = RbEncoding.new(encoding) }
+      end
+    end
```

Comment:
Looks good to me.
We might want to make `RbEncoding.new` private to avoid accidentally calling it.

Note that with my upcoming concurrent collections,
```ruby
@cache = Hash.new { |h,k| h[k] = RbEncoding.new(k) }
```
would synchronize as needed on its own and have put-if-absent semantics inside that block.

Reply:
I very much look forward to that :-)